### DISCUSSION TITLE
close connection immediately and blacklist

### DISCUSSION THREAD

**Discussion by lithbitren** [tokio-rs/axum#1620] [Q&A] [answered]
if client's connection is probably attacking, how can I close it immediately and insert the address into a dynamic blacklist?

pseudocode:

```
    let listener = TcpListener::bind(...).await?;

    while let (socket, addr) = listener.accept().await? {

        if blacklist.contains(&addr) {
            continue;
        }

        tokio::spawn(async move {

            ...

            if connection_is_attacking() {
                blacklist.insert(addr);
                connection.close_immediately();
            }

            ...

        });
    }
```

**Comment by davidpdrsn** [comment, marked as answer]
I don't know exactly, haven't done this before. But you can probably use hyper's low level connection API (https://docs.rs/hyper/latest/hyper/server/conn/index.html) and do a `tokio::select!` between handling the connection and aborting. To close the connection you just `drop` it.
